test/types/bulkCreate.ts

```diff
@@ -0,0 +1,76 @@
+import {
+  Model,
+  InferAttributes,
+  CreationOptional,
+  InferCreationAttributes,
+} from '@sequelize/core';
+import { sequelize } from './connection';
+import type { MakeNullishOptional } from '@sequelize/core/types/utils';
+
+class TestModel extends Model<
+  InferAttributes<TestModel>,
+  InferCreationAttributes<TestModel>
+> {
+  declare id: CreationOptional<number>;
+  declare testString: CreationOptional<string | null>;
+  declare testEnum: CreationOptional<'d' | 'e' | 'f' | null>;
+}
+
+type wat = InferCreationAttributes<TestModel>;
+
+sequelize.transaction(async trx => {
+  type B = InferCreationAttributes<TestModel>;
+
+  type C = MakeNullishOptional<B>;
+  const badItems: Array<
+    MakeNullishOptional<InferCreationAttributes<TestModel>>
+  > = [
+    {
+      // @ts-expect-error this isn't an enum value
+      testEnum: 'eafe',
+    },
+    {
+      // @ts-expect-error testString should be a string or null
+      testString: 324,
+    },
+  ];
+
+  const newItems: Array<
+    MakeNullishOptional<InferCreationAttributes<TestModel>>
```

Comment:
`Infer*` should only be used during model definition for portability, but the following type is equivalent to what you're doing:

```suggestion
    CreationAttributes<TestModel>
```